Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -637,40 +665,41 @@ extension BrowserViewController: URLBarDelegate {
         }
     }
 
-    func urlBarDidLongPressReaderMode(urlBar: URLBarView) {
-        if let tab = tabManager.selectedTab {
-            if var url = tab.displayURL {
-                if let absoluteString = url.absoluteString {
-                    let result = profile.readingList?.createRecordWithURL(absoluteString, title: tab.title ?? "", addedBy: UIDevice.currentDevice().name) // TODO Check result, can this fail?
-                    // TODO Followup bug, provide some form of 'this has been added' feedback?
-                }
+    func urlBarDidLongPressReaderMode(urlBar: URLBarView) -> Bool {
+        if let tab = tabManager.selectedTab,
+               url = tab.displayURL,
+               absoluteString = url.absoluteString,
+               result = profile.readingList?.createRecordWithURL(absoluteString, title: tab.title ?? "", addedBy: UIDevice.currentDevice().name)
+        {
+            switch(result) {
```

Comment:
Btw. here I broke the syntax guidelines, but IMHO for good reason - including the beginning brace for `if` on the same line as `result = ...` would seem to place the `switch` at a lower indent level than the previous line. The way I did it, it is clear that the `switch` is inside the `if` since it is at a higher indent level than the preceding line (containing sole `{` on the same indent level as the `if`).